src/utils/validation.ts

```diff
@@ -32,31 +30,21 @@ export const testVenueByNameExists = async (value: unknown) => {
 export const testWorldBySlugExists = (worldId: WorldId) => async (
   value: unknown
 ) => {
-  // @debt unsure how test logic at the bottom should work for creating new world
-  // but keep this check and throw of error, TS just declares worldId defined, doesn't make it so
-  if (!worldId) {
-    throw new SparkleFetchError({
-      message: `Invalid worldId: ${String(worldId)} for value: ${String(
-        value
-      )}`,
-      args: {
-        worldId,
-        value,
-      },
-    });
-  }
-
   const slug = createSlug(value);
   if (!slug) return false;
 
   // @debt Replace with a function from api/worlds
-  const snap = await firebase
+  let path = firebase
     .firestore()
     .collection("worlds")
     .where("slug", "==", slug)
-    .where("isHidden", "==", false)
-    .where(firebase.firestore.FieldPath.documentId(), "!=", worldId)
-    .get();
+    .where("isHidden", "==", false);
+
+  if (worldId) {
+    path = path.where(firebase.firestore.FieldPath.documentId(), "!=", worldId);
+  }
```

Comment:
maybe we should be using something like this?
`
const path = firebase .firestore().collection("worlds").where("slug", "==", slug).where("isHidden", "==", false);
const result = worldId ? path.where(firebase.firestore.FieldPath.documentId(), "!=", worldId) : path;
`
Just so we avoid this unnecessary `path` mutation

Feel free to rename `result` into something else, I'm bad at naming :)

Reply:
@alexey-petrov, done!
https://github.com/sparkletown/sparkle/pull/2936/commits/28189cf95ab26774051c2503b83fd0f499fda1ba